### PR TITLE
RN-65 RN: Show channel intro message

### PR DESCRIPTION
**Pull request by csduarte**
#### Summary
This PR adds channel intros to the beginning of channels.

#### Ticket Link
https://mattermost.atlassian.net/browse/RN-65

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Added or updated unit tests (required for all new features)
- [ ] All new/modified APIs include changes to [mattermost-redux](https://github.com/mattermost/mattermost-redux) (please link)
- [x] Has UI changes
- [x] Includes text changes and localization file updates

#### Device Information
This PR was tested on:
iPhone 6 iOS 10
iPhone 7 iOS 10
Galaxy s7 - Android 6.0 API 23 

#### Screenshots
simulator screen shot apr 13 2017 7 53 10 am
simulator screen shot apr 13 2017 7 53 46 am
simulator screen shot apr 13 2017 7 53 59 am
simulator screen shot apr 13 2017 8 04 29 am
simulator screen shot apr 13 2017 8 05 54 am

@jarredwitt

### CODE REVIEW

File: app/components/channel_intro/channel_intro.js [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+    }
+
+    buildGMContent = () => {
+        const {intl, theme} = this.props;
+        const style = getStyleSheet(theme);
+
+        return (
+            <Text style={style.message}>
+                {intl.formatMessage({
+                    id: 'intro_messages.GM',
```
Comment (enahum) [inline app/components/channel_intro/channel_intro.js]: this message is wrong, im adding a PR to `platform` that fixes this

File: app/components/channel_intro/channel_intro.js [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+        const mainMessage = intl.formatMessage({
+            id: 'intro_messages.creator',
+            defaultMessage: 'This is the start of the {name} {type}, created by {creator} on {date}.'
+        }, {
+            name: currentChannel.display_name,
+            creator: creatorName,
+            date,
+            type: intl.formatMessage({
+                id: 'intro_messages.group',
+                defaultMessage: 'private group'
```
Comment (enahum) [inline app/components/channel_intro/channel_intro.js]: the defaultMessage here should be `private channel`

File: app/components/channel_intro/channel_intro.js [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+            })
+        });
+
+        const onlyInvitedMessage = intl.formatMessage({
+            id: 'intro_messages.onlyInvited',
+            defaultMessage: ' Only invited members can see this private group.'
```
Comment (enahum) [inline app/components/channel_intro/channel_intro.js]: same here

File: assets/base/i18n/en.json [hunk 4]
```diff
@@ -1537,6 +1537,7 @@
   "integrations.outgoingWebhook.title": "Outgoing Webhook",
   "integrations.successful": "Setup Successful",
   "intro_messages.DM": "This is the start of your direct message history with {teammate}.<br />Direct messages and files shared here are not shown to people outside this area.",
+  "intro_messages.GM": "This is the very beginning of your group message.",
```
Comment (enahum) [inline assets/base/i18n/en.json]: this one we should remove it from here, we should be adding only those that have a `mobile.` prefix.

File: app/components/channel_intro/channel_intro.js (hunk 1 above)
Comment (jarredwitt) [inline app/components/channel_intro/channel_intro.js]: Did this get added?

File: assets/base/i18n/en.json (hunk 4 above)
Comment (jarredwitt) [inline assets/base/i18n/en.json]: Guessing this will be added on the platform side?

Comment (enahum) [inline assets/base/i18n/en.json]: yeah just got merged into master https://github.com/mattermost/platform/pull/6116

**Comment by jarredwitt** [comment]
@enahum changes made.

File: app/components/channel_intro/channel_intro.js [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+    }
+
+    buildGMContent = () => {
+        const {intl, theme} = this.props;
+        const style = getStyleSheet(theme);
+
+        return (
+            <Text style={style.message}>
+                {intl.formatMessage({
+                    id: 'intro_messages.group_message',
```
Comment (enahum) [inline app/components/channel_intro/channel_intro.js]: @jarredwitt I think we are missing another option here that in platform is `intro_messages.GM`

Comment (jarredwitt) [inline app/components/channel_intro/channel_intro.js]: According to the UI design it just says `this is the very beginning of your group message.` We already have the names listed above the message. 

image

Comment (enahum) [inline app/components/channel_intro/channel_intro.js]: oh makes sense

**Pull request merged**